### DISCUSSION TITLE
how to use speed-measure-webpack-plugin with Nextjs

### DISCUSSION THREAD

**Discussion by KingMatrix1989** [vercel/next.js#33678] [Help]
Hi. i want to use SMP with nextjs config. please help

**Comment by sagaekeiga** [comment]
@skmohammadi 

next.config.js
```js
const SpeedMeasurePlugin = require('speed-measure-webpack-plugin')
const smp = new SpeedMeasurePlugin()

const moduleExports = {
  webpack(config) {
    return smp.wrap(config)
  },
}
```

```
yarn dev
```


![image](https://user-images.githubusercontent.com/19757050/158003142-a0782566-55cf-4588-a143-8cf1d7d607eb.png)

**Reply by duuliy** [reply]
> ```js
> moduleExports
> ```

the next version is 12.0.10, and an error will be reported
error - ./node_modules/next/dist/build/webpack/loaders/next-client-pages-loader.js?absolutePagePath=private-next-pages%2F_app&page=%2F_app!
TypeError: Cannot read properties of undefined (reading 'traceChild')

**Reply by BettyEE** [reply]
请问这个有解决吗？

**Reply by satyamgaur** [reply]
@duuliy found any workaround for this?

**Comment by zlm11** [comment]
The smp should be used like this:
```
const SpeedMeasurePlugin = require('speed-measure-webpack-plugin');
const smp = new SpeedMeasurePlugin();
webpack: (config) => {
    config.plugins.push(smp);
    return config;
},
```
@BettyEE @skmohammadi @satyamgaur

**Reply by rco7o9** [reply]
Hey @zlm11 , I tried the above in my NextJS application and the error went away but as we have not used the wrap method, SMP is not able to provide stats for the plugins and only shows stats for the loaders. Any workaround for this ?
